File: src/main/java/tv/codealong/tutorials/various/audit/java/HttpSenderConfiguration.java
package tv.codealong.tutorials.various.audit.java;


import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

public class HttpSenderConfiguration extends SenderConfiguration {
    public static final String NODE_ID_HEADER = "X-Node-ID";
    public static final String BASE_URL_PROPERTY = "base-url";
    public static final String FLUSH_METRIC_PERIOD_PROPERTY = "flush-metric-period";
    public static final String FLUSH_METRIC_TIME_UNIT_PROPERTY="flush-metric-time-unit";
    public static final String RETRY_PERIOD_MS_PROPERTY = "retry-period-ms";
    public static final String RETRY_MAX_PERIOD_MS_PROPERTY = "retry-max-period-ms";
    public static final String RETRY_MAX_ATTEMPTS_PROPERTY = "retry-max-attempts";
    public static final String MAX_TOTAL_HTTP_CONNECTIONS = "max-total-http-connections";
    public static final String DEFAULT_MAX_PER_ROUTE = "default-max-per-route";
    private Map<String, Object> properties;
    private Map<String, HttpRoute> routes;

    public void validate() {
        this.checkRequiredProperty("base-url");
        this.checkRequiredBaseHeader("X-Node-ID");
        List<String> missingRoutes = (List) this.getRouteResolvers ().stream().map(TransportRouteConfig::getRoute).filter((route)
        return this.routes.containsKey(route);
    }).collect (Collectors.toList());
if (!missingRoutes.isEmpty()) {
-> {
            throw new IllegalStateException(String.format("Unable to find routes %s for route resolvers of HttpSender %s", missingRoutes, this.getName()));
        }
    }
    public void setRoutes (List<HttpRoute> routes) {
        this.routes = (Map) routes.stream().collect(Collectors.toMap (HttpRoute::getName, (httpRoute)));
        return httpRoute;
}

File: src/main/java/tv/codealong/tutorials/various/audit/HttpSenderConfiguration.kt
package tv.codealong.tutorials.various.audit

import tv.codealong.tutorials.various.audit.java.HttpRoute

/**
 * Содержит базовую конфигурацию HTTP-клиента
 *
 * Управляет маршрутизацией через routeResolvers
 *
 * Хранит базовые заголовки и свойства
 */
class HttpSenderConfiguration {
    companion object {
        const val BASE_URL_PROPERTY = "base.url"
    }

    var name: String? = null
    var baseHeaders: Map<String, String> = emptyMap()
    var routeResolvers: List<TransportRouteConfig> = emptyList()
    var properties: Map<String, String> = emptyMap()
    private val routes: MutableMap<String, HttpRoute> = mutableMapOf()

    fun setRoutes(routes: List<HttpRoute>) {
        this.routes.clear()
        routes.forEach { this.routes[it.name] = it }
    }

    fun getRoute(name: String): HttpRoute? = routes[name]

    fun validate() {
        require(name?.isNotBlank() == true) { "Sender name must be specified" }
        require(properties.containsKey(BASE_URL_PROPERTY)) { "Base URL must be specified" }
    }
}

/**
 * Определяет конечные точки API
 *
 * Содержит имя маршрута, путь и метод
 *
 * Может включать специфичные для маршрута заголовки
 */
//data class HttpRoute(
//    val name: String,
//    val path: String,
//    val method: HttpMethod,
//    val headers: Map<String, String> = emptyMap()
//) {
//    init {
//        require(name.isNotBlank()) { "Route name must not be blank" }
//        require(path.isNotBlank()) { "Path must not be blank" }
//    }
//}
//
//enum class HttpMethod {
//    GET, POST, PUT, DELETE, PATCH, HEAD, OPTIONS
//}

//можно и такой класс сделать:
//class Request {
//    enum class HttpMethod {
//        GET, POST, PUT, DELETE, PATCH, HEAD, OPTIONS
//    }
//}